### PR TITLE
Update sparql-gp.js, less DDoS

### PR DESCRIPTION
**Pull request by kltm**
For https://github.com/geneontology/api-gorest/issues/3
Trying to bring on @balhoff 's changes to the experimental codebase (https://github.com/geneontology/api-gorest/issues/3#issuecomment-974166580)

### CODE REVIEW

File: queries/sparql-gp.js [hunk 1]
```diff
@@ -70,80 +70,65 @@ module.exports = {
...
-          negatively_regulates: positively_regulates: directly_regulates: directly_positively_regulates: directly_negatively_regulates:
-          directly_activates: indirectly_activates: directly_inhibits: indirectly_inhibits: transitively_provides_input_for: 
-          immediately_causally_upstream_of: directly_provides_input_for: }
-            
+      PREFIX hint: <http://www.bigdata.com/queryHints#>
+      SELECT DISTINCT ?gocam ?title
+      WHERE {
+        GRAPH ?gocam {  
+          # Inject gene product ID here
+          ?gene rdf:type <http://identifiers.org/rgd/1586460> .
```
Comment (dustine32) [inline queries/sparql-gp.js]: I spot a hard-coded ID here. I think this
```
?gene rdf:type <http://identifiers.org/rgd/1586460> .
```
should be replaced by 
```
?gene rdf:type <` + id + `> .

```

File: queries/sparql-gp.js [hunk 2]
```diff
@@ -70,80 +70,65 @@ module.exports = {
...
-          
-         FILTER(?gp1 = <` + id + `> || ?gp2 = <` + id + `> || ?gp3 = <` + id + `>)
-      
+          ?ind1 ?causal ?ind2 .
+          ?ind2 ?causal2 ?ind3 .
+          ?gpnode1 rdf:type ?gp1 .
+          ?gpnode2 rdf:type ?gp2 .
+          ?gpnode3 rdf:type ?gp3 .
+          # Inject gene product ID here
+          FILTER(?gp1 = <http://identifiers.org/rgd/1586460> || ?gp2 = <http://identifiers.org/rgd/1586460> || ?gp3 = <http://identifiers.org/rgd/1586460>)
```
Comment (dustine32) [inline queries/sparql-gp.js]: Same here:
```
FILTER(?gp1 = <http://identifiers.org/rgd/1586460> || ?gp2 = <http://identifiers.org/rgd/1586460> || ?gp3 = <http://identifiers.org/rgd/1586460>)
```
to:
```
FILTER(?gp1 = <` + id + `> || ?gp2 = <` + id + `> || ?gp3 = <` + id + `>)
```

**Review by dustine32 — CHANGES_REQUESTED** [review]
@kltm I can try applying these changes unless you beat me too it!

**Comment by dustine32** [comment]
@kltm I just pushed the change. Gonna merge in like 5 min here then we can see if this dang thing works!

**Comment by kltm** [comment]
Whoops--sorry about that.

**Review by dustine32 — APPROVED** [review]

**Comment by dustine32** [comment]
@kltm Heh, no prob! Thanks!

**Pull request merged**